File: node-red-data/sequelize/models/device-state.js
'use strict';

const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class DeviceState extends Model {

    static associate(models) {
      DeviceState.belongsTo(models.Device, { foreignKey: 'deviceId', as: 'device' });
    }
  }
  DeviceState.init({
    deviceId: DataTypes.INTEGER,
    state: DataTypes.JSON,
  }, {
    sequelize,
    modelName: 'DeviceState',
    tableName: 'device_state',
    freezeTableName: true,
  });



  return Device;
};

File: flow-space-reporting/app/data_models/device_state.py
from typing import TYPE_CHECKING, Optional

from sqlalchemy import Integer, JSON, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.data_models.base import Base, TimestampMixin

if TYPE_CHECKING:
    from app.data_models.device import Device

class DeviceState(Base, TimestampMixin):
    __tablename__ = "device_state"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    device_id: Mapped[int | None] = mapped_column(
        ForeignKey("device.id"), name="deviceId"
    )
    state: Mapped[dict | None] = mapped_column(JSON)

    device: Mapped[Optional["Device"]] = relationship(back_populates="states")
